src/agentscope/agents/rag_agents.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+"""
+This example shows how to build an agent with RAG
+with LlamaIndex.
+
+Notice, this is a Beta version of RAG agent.
+"""
+
+from typing import Optional, Any
+from loguru import logger
+
+from agentscope.agents.agent import AgentBase
+from agentscope.message import Msg
+from agentscope.rag import Knowledge
+
+
+CHECKING_PROMPT = """
+                Is the retrieved content relevant to the query?
+                Retrieved content: {}
+                Query: {}
+                Only answer YES or NO.
+                """
+
+
+class LlamaIndexAgent(AgentBase):
+    """
+    A LlamaIndex agent build on LlamaIndex.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        sys_prompt: str,
+        model_config_name: str,
+        knowledge_list: list[Knowledge] = None,
+        memory_config: Optional[dict] = None,
+        rag_config: Optional[dict] = None,
```

Comment:
declare a `RagConfig` class may be better

Reply:
done